Investigate heisenbug with db AddBatch()

#### Issue description

A heisenbug which causes gowarc to panic at db AddBatch()

#### Steps to reproduce the issue

Currently unknown, but indexing many warcs increase the odds


#### What's the expected result?

All warcs are either indexed or some report errors i.e invalid file content 


#### What's the actual result?

Some warcs causes gowarc to panic at random


#### Additional details / screenshot

Panic stack trace
```txt
panic: runtime error: invalid memory address or nil pointer dereference                       
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa0a59c]                       
goroutine 23 [running]:                                                                       
github.com/nlnwa/gowarc/pkg/index.(*Db).AddBatch.func1(0xc517e22980, 0x450001, 0xc517e22980)  
        /build/pkg/index/db.go:253 +0x8c                                                      
github.com/dgraph-io/badger/v2.(*DB).Update(0xc0001ce800, 0xc78fa4bf50, 0x0, 0x0)             
        /go/pkg/mod/github.com/dgraph-io/badger/v2@v2.0.2/txn.go:696 +0x94                    
github.com/nlnwa/gowarc/pkg/index.(*Db).AddBatch(0xc0001d7ab0, 0xc4fd83e000, 0x2711, 0x2711)  
        /build/pkg/index/db.go:251 +0x100                                                     
github.com/nlnwa/gowarc/pkg/index.NewIndexDb.func1(0xc0001d7ab0, 0xc0000e8e40)                
        /build/pkg/index/db.go:79 +0x71                                                       
created by github.com/nlnwa/gowarc/pkg/index.NewIndexDb                                       
        /build/pkg/index/db.go:77 +0x31a  
```

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/nlnwa/gowarc
 go 1.13
 
 require (
-	github.com/dgraph-io/badger/v2 v2.0.2
+	github.com/dgraph-io/badger/v2 v2.2007.2
 	github.com/fsnotify/fsnotify v1.4.7
 	github.com/golang/protobuf v1.4.0-rc.4
 	github.com/gorilla/handlers v1.4.2

